src/Dictionaries/DictionaryFactory.h

```diff
@@ -54,14 +54,16 @@ class DictionaryFactory : private boost::noncopyable
         bool created_from_ddl)>;
 
     bool isComplex(const std::string & layout_type) const;
+    bool convertToComplex(std::string & layout_type) const;
```

Comment:
Consider to add comment for this function. It can be tricky for client to understand that argument will be modified if there is complex layout type.

Reply:
yes, I will add some comments